pre_commit_hooks/insert_license.py

```diff
@@ -168,14 +169,18 @@ def process_files(args, changed_files, todo_files, license_info: LicenseInfo):
                 fuzzy_ratio_cut_off=args.fuzzy_ratio_cut_off,
             )
         if license_header_index is not None:
-            if license_found(remove_header=args.remove_header,
-                             update_year_range=args.use_current_year,
-                             license_header_index=license_header_index,
-                             license_info=license_info,
-                             src_file_content=src_file_content,
-                             src_filepath=src_filepath,
-                             encoding=encoding):
-                changed_files.append(src_filepath)
+            try:
+                if license_found(remove_header=args.remove_header,
+                                 update_year_range=args.use_current_year,
+                                 license_header_index=license_header_index,
+                                 license_info=license_info,
+                                 src_file_content=src_file_content,
+                                 src_filepath=src_filepath,
+                                 encoding=encoding):
+                    changed_files.append(src_filepath)
+            except RuntimeError as error:
+                print(error)
```

Comment:
I'm not fan of this error-catching strategy:
you have zero ensurance that the error caught will be related to the year update operation!
Please either create a custom exception class or move this try-catch around the call to `try_update_year_range()` and add a boolean return value to `license_found()`

Reply:
I went with a custom exception class. I'm not a fan of returning just two booleans from a single function.